xdsl/dialects/riscv.py

```diff
@@ -346,13 +351,55 @@ def parse(cls, parser: Parser) -> Self:
             regions=regions,
         )
 
+    @classmethod
+    def parse_args(cls, parser: Parser) -> Sequence[UnresolvedOperand]:
+        if arg := parser.parse_optional_unresolved_operand():
+            args = [arg]
+            while parser.parse_optional_punctuation(",") and (
+                arg := parser.parse_optional_unresolved_operand()
+            ):
+                args.append(arg)
+            return args
+        return []
+
+    @classmethod
+    def parse_attributes(cls, parser: Parser) -> Mapping[str, Attribute]:
```

Comment:
Why do we redefine these things here? Can we just call `parser.parse_optional_attr_dict` directly?

Reply:
`parse_attributes` will be override by classes which implement custom attribute syntax. By default it uses generic attribute dict. See [here](https://github.com/xdslproject/xdsl/pull/1318/files#diff-c2e583c286e17ace72eadfcf055304a684874cd8bfbb4c2b51ff7d9d21ff7be4R2141) for `riscv.DirectiveOp` example.